tests/components/homekit_controller/test_config_flow.py

```diff
@@ -934,103 +652,48 @@ def test_valid_pairing_codes(pairing_code):
     }
 
 
-async def test_unignore_works(hass):
+async def test_unignore_works(hass, controller):
     """Test rediscovery triggered disovers work."""
-    discovery_info = {
-        "name": "TestDevice",
-        "address": "127.0.0.1",
-        "port": 8080,
-        "md": "TestDevice",
-        "pv": "1.0",
-        "id": "00:00:00:00:00:00",
-        "c#": 1,
-        "s#": 1,
-        "ff": 0,
-        "ci": 0,
-        "sf": 1,
-    }
-
-    pairing = mock.Mock(pairing_data={"AccessoryPairingID": "00:00:00:00:00:00"})
-    pairing.list_accessories_and_characteristics = asynctest.CoroutineMock(
-        return_value=[
-            {
-                "aid": 1,
-                "services": [
-                    {
-                        "characteristics": [
-                            {"type": "23", "value": "Koogeek-LS1-20833F"}
-                        ],
-                        "type": "3e",
-                    }
-                ],
-            }
-        ]
-    )
-
-    finish_pairing = asynctest.CoroutineMock()
-
-    discovery = mock.Mock()
-    discovery.device_id = "00:00:00:00:00:00"
-    discovery.info = discovery_info
-    discovery.start_pairing = asynctest.CoroutineMock(return_value=finish_pairing)
+    device = setup_mock_accessory(controller)
 
-    finish_pairing.return_value = pairing
-
-    flow = _setup_flow_handler(hass)
-
-    flow.controller.pairings = {"00:00:00:00:00:00": pairing}
-    flow.controller.discover_ip = asynctest.CoroutineMock(return_value=[discovery])
-
-    flow.controller.find_ip_by_device_id = asynctest.CoroutineMock(
-        return_value=discovery
+    # Device is unignored
+    result = await hass.config_entries.flow.async_init(
+        "homekit_controller",
+        context={"source": "unignore"},
+        data={"unique_id": device.device_id},
     )
-
-    result = await flow.async_step_unignore({"unique_id": "00:00:00:00:00:00"})
     assert result["type"] == "form"
     assert result["step_id"] == "pair"
-    assert flow.context == {
+    assert get_flow_context(hass, result) == {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5066052)